thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -529,13 +529,17 @@ private void decodeAndInvoke(
                 decodedReq = toRpcRequest(f.serviceType(), header.name, args);
                 ctx.logBuilder().requestContent(decodedReq, new ThriftCall(header, args));
             } catch (Exception e) {
-                // Failed to decode the invocation parameters.
-                logger.debug("{} Failed to decode Thrift arguments:", ctx, e);
-
-                final TApplicationException cause = new TApplicationException(
-                        TApplicationException.PROTOCOL_ERROR, "failed to decode arguments: " + e);
-
-                handlePreDecodeException(ctx, httpRes, cause, serializationFormat, seqId, methodName);
+                final TApplicationException cause;
+                if (ctx.config().verboseResponses()) {
+                    cause = new TApplicationException(
+                            TApplicationException.PROTOCOL_ERROR, "failed to decode arguments: " + e);
+                } else {
+                    // The exception could have sensitive information such as the required field.
+                    // So we don't include the cause message unless verboseResponses returns true.
+                    cause = new TApplicationException(TApplicationException.PROTOCOL_ERROR,
+                                                      "failed to decode arguments for " + header.name);
+                }
+                handleException(ctx, httpRes, serializationFormat, seqId, f, cause);
```

Comment:
Q: Don't we need to call `handlePreDecodeException` anymore?

Reply:
Yeah, I guess we don't have to use `exceptionHandler` for this case. Let me revert to using `handlePreDecodeException `. Thanks!